primitives/core/src/crypto.rs

```diff
@@ -112,15 +114,15 @@ pub enum SecretStringError {
 /// a new secret key from an existing secret key and, in the case of `SoftRaw` and `SoftIndex`
 /// a new public key from an existing public key.
 #[derive(Copy, Clone, Eq, PartialEq, Hash, Debug, Encode, Decode)]
-#[cfg(feature = "full_crypto")]
+#[cfg(any(feature = "full_crypto", feature = "serde"))]
```

Comment:
Wasn't sure what's better here: Introduce `any` or leave away the conditional import completely. Let me know what you prefer.

Reply:
It looks fine for me. We need it for both features.